### DISCUSSION TITLE
Where did you learn to do all of this?

### DISCUSSION THREAD

**Discussion by evanwheelrr** [dorssel/cryptoref#4] [General]
I have been looking for a encrytion algorithm for my crypter, I was scammed a while back for a fake crypter i never got code for so i decided to make my own, I ended up getting as far as generating exe's to a custom path with a custom name all inside a GUI. I have lately been looking for a encryption algorithm to bypass windows antivirus so I can get meterpreter payloads to work without being detected, the code you write I am not sure what it does but believe it is differnt encryption algorithms. I would love to hear your story of where you started out and the path you took to gain such knowledge. I was also reading a pdf of differnt bypass techneques and it linked me to your github

**Comment by dorssel** [comment]
This particular library is a *reference* implementation, following the referenced standards as closely as possible. It is my personal study project. I learned most of this from mathematics courses in university and from books. If you want practical advise: there is an important wisdom to remember "do not implement your own crypto". At least not for stuff that matters / is important. Unless you are a cryptographer.

**Reply by evanwheelrr** [reply]
where might I find the original author for the code?

**Reply by dorssel** [reply]
The code in this repository? That's mine. Or do you mean the original developers of these algorithms? I wouldn't know, but they are probably referenced in the NIST standards.

**Comment by evanwheelrr** [comment]
Right, I was curious how you learned to write that type of code; are you
currently working on any bypass techniques? or do you know where I can
look? I started to use c# to be able to suspend a process, but it used a
module and was very limited. The goal was to be able to modify the code of
a running application, and I was also considering combining the techniques
you wrote. below I haveBelow, attached the pdf's Ipdfs was reading. I am
trying to know how to understand these pdf's more, this stuff doesn't seem
to be readily available on the internet

**Reply by dorssel** [reply]
Self taught, just by examining lots of other code. I'm sorry I can't help you further.

**Comment by evanwheelrr** [comment]
Ok, thanks for the reply.
